### PR TITLE
feat(http): Add option for HTTP listen host and port

### PR DESCRIPTION
**Pull request by jbuck**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -97,9 +103,16 @@ func parseArgsAndLoadConf() {
 	if err != nil {
 		log.Fatal(err)
 	}
+
+	if conf.Host == "" {
+		conf.Host = host
+	}
+	if conf.Port == 0 {
+		conf.Port = port
+	}
```
Comment (jmhodges) [inline main.go:112]: Could we just use the config file for this? I'm not sure if it's important to support both (and I had to read the code twice to make sure I understand what was happening here.)

File: main_test.go [hunk 2]
```diff
@@ -356,6 +362,9 @@ func Test_prepareServerOptions(t *testing.T) {
 }
 
 func Test_preparedServer(t *testing.T) {
+	// reset global http server
+	http.DefaultServeMux = new(http.ServeMux)
```
Comment (jmhodges) [inline main_test.go:366]: No, please don't do modify global state for this. Just inject the handlers instead.

**Review by jmhodges — CHANGES_REQUESTED** [review]

File: main_test.go (hunk 2 above)
Comment (jmhodges) [reply main_test.go:366]: (By "inject", I mean "pass them as an argument")

File: main.go (hunk 1 above)
Comment (jbuck) [reply main.go]: Done, removed the flag

File: main_test.go [hunk 3]
```diff
@@ -329,14 +329,30 @@ func Test_validateBaseURL(t *testing.T) {
...
+
 func Test_preparedServer(t *testing.T) {
 	// For the purpose of testing - ensure we're using IPv4.
 	conf.BaseURL = "http://127.0.0.1:8000/"
 
 	testServer := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprintln(w, "Hello, client")
 	}))
-	testServer.Config = prepareServer()
+	testServer.Config = prepareServer("", 8080)
```
Comment (jmhodges) [inline main_test.go:355]: Pass 0 in here so that we don't need to leave port 8080 open just to run the unit tests.

**Review by jmhodges — APPROVED** [review]
One not-required change

File: main_test.go (hunk 3 above)
Comment (jmhodges) [reply main_test.go:355]: It might be that httptest ignores the Addr setting. I can't recall

**Review by jmhodges — APPROVED** [review]

**Pull request merged**